docs/reviews/SINK_CHECKLIST.md

```diff
@@ -0,0 +1,32 @@
+This is a checklist to use whilst reviewing code for Vector's sinks.
+
+## Logic
+
+- [ ] Does it work? Do you understand what it is supposed to be doing?
+- [ ] Does the retry logic make sense?
+- [ ] Are they emitting the correct metrics?
+- [ ] Are the tests testing that they are emitting the correct metrics?
```

Comment:
Can these be enforced by traits? At least that there is an emission - it could still be the wrong value.

Reply:
For V2 sinks yes, at least we can get quite close providing they are using the driver correctly. It is still worth testing to make sure.